odl/contrib/theano/layer.py

```diff
@@ -17,60 +17,260 @@
 import theano
 import numpy as np
 
+from odl.solvers import Functional
+
 
 __all__ = ('TheanoOperator',)
 
 
 class TheanoOperator(theano.Op):
-    """Wraps ODL operator to use with theano.
+
+    """Wrap an ODL operator as a Theano operator.
 
     The operator works with theanos automatic differentiation if the operator
     implements `Operator.derivative` and `Operator.adjoint`.
     """
 
-    # Properties used by theano to check for equality
+    # Properties used by theano for __eq__ and __hash__
     __props__ = ('operator',)
 
     def __init__(self, operator):
         """Initialize an instance.
 
         Parameters
         ----------
-        operator : Operator
+        operator : `Operator`
             The operator that should be wrapped, must map `FnBase` spaces to
             `FnBase` spaces.
+
+        Examples
+        --------
+        Make a vector-to-vector operator:
+
+        >>> space = odl.rn(3)
+        >>> matrix = np.array([[1, 0, 1],
+        ...                    [0, 1, 1]], dtype=float)
+        >>> op = odl.MatrixOperator(matrix, domain=space)
+        >>> matrix_op = TheanoOperator(op)
+        >>> matrix_op.operator is op
+        True
+
+        Create a functional, i.e., an operator with scalar output:
+
+        >>> space = odl.rn(3)
+        >>> functional = odl.solvers.L2NormSquared(space)
+        >>> func_op = TheanoOperator(functional)
+        >>> x = theano.tensor.dvector()
+        >>> apply = func_op.make_node(x)
```

Comment:
func_op(x) does not work?

Reply:
It should, doctest seems incomplete regarding that. Probably I just wanted to see the "scalar" word in the output.

Reply:
The result is `TheanoOperator{operator=L2NormSquared(rn(3))}.0`, which is not bad, but I'm not sure if it's good enough for a doctest.